### DISCUSSION TITLE
Visible/Hide TextInput when choose select item

### DISCUSSION THREAD

**Discussion by iranandroid** [filamentphp/filament#3891] [Help]
Hi
I have a ```Select``` field with two option: banner and event. now wanna when select, evet option show ```DatePicker``` field and when select banner option hide ```DatePicker``` and show ```TextInput```, but I don't  know how do it

**Comment by Z3d0X** [comment]
```php
Select::make('type')
    ->options([ 
        'event' => 'Event',
        'banner' => 'Banner'
    ])
    ->reactive(),

DatePicker::make('event')
    ->visible(fn (\Closure $get) => $get('type') === 'event'),

TextInput::make('banner')
    ->visible(fn (\Closure $get) => $get('type') === 'banner'),
```
https://filamentphp.com/docs/2.x/forms/advanced#dependant-fields--components

**Reply by nitinatvaluelabs** [reply]
Hi, I tried this but not working. Do you know why?

Edit: Working now. You need to make sure that the schema is `reactive()`
